omnibus/dependabot-omnibus.gemspec

```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |spec|
   spec.homepage     = common_gemspec.homepage
   spec.license      = common_gemspec.license
 
-  spec.required_ruby_version = ">= 2.7.0"
+  spec.required_ruby_version = ">= 3.1.0"
```

Comment:
I'm a little surprised we need to pin this here, I would have thought we'd inherit this pin from https://github.com/dependabot/dependabot-core/pull/5447/files#diff-e843e646a7311fb218cf05845818b31d84bc04c0742f0567395c0c289cacf72cR21

Can anyone who's more familiar with ruby gemspec inheritance comment on whether we need to keep this pin? 

cc @deivid-rodriguez @jurre

Reply:
Ah NVM, I see it was added to pacify Rubocop here: https://github.com/dependabot/dependabot-core/commit/37cff08adfa60a6cfa251b678b815f0c046543d1

I verified that removing the pin causes Rubocop to complain again:
```
Offenses:

dependabot-omnibus.gemspec:1:1: C: Gemspec/RequiredRubyVersion: required_ruby_version should be specified.
```